.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1 +1,30 @@
-<!-- Please provide a clear and meaningful description to the CHANGELOG.md file if this PR contributes some significant changes -->
+<!--
+Thank you for your Pull Request. Please provide a description and review
+the requirements below.
+
+Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md
+-->
+
+#### What it does
+<!-- Include relevant issues and describe how they are addressed. -->
+
+#### How to test
+<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify perf improvements.  -->
+
+#### Review Checklist
```

Comment:
I would go with `Review checklist` as we do not use a capital for the other titles either.

[minor]: perhaps you could start each bullet point with a capital, and finish the sentence with a period. But it is up to you.